assessment/sources/mysql/infoschema_test.go

```diff
@@ -0,0 +1,841 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mysql
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/DATA-DOG/go-sqlmock"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/assessment/utils"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/internal"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/schema"
+	"github.com/stretchr/testify/assert"
+)
+
+// Helper to create InfoSchemaImpl with mock DB
+func newTestInfoSchemaImpl(t *testing.T) (InfoSchemaImpl, sqlmock.Sqlmock) {
+	db, mock, err := sqlmock.New(sqlmock.QueryMatcherOption(sqlmock.QueryMatcherRegexp))
+	assert.NoError(t, err)
+	return InfoSchemaImpl{Db: db, DbName: "test_db"}, mock
+}
+
+func TestInfoSchemaImpl_GetTableInfo(t *testing.T) {
+	tableQueryRegex := `SELECT TABLE_COLLATION, SUBSTRING_INDEX\(TABLE_COLLATION, '_', 1\) as CHARACTER_SET FROM INFORMATION_SCHEMA\.TABLES WHERE TABLE_SCHEMA = \? AND TABLE_NAME = \?`
+	columnQueryRegex := `SELECT c\.column_type, c\.extra, c\.generation_expression\s+FROM information_schema\.COLUMNS c\s+where table_schema = \? and table_name = \? and column_name = \?\s+ORDER BY c\.ordinal_position;`
+
+	type testCase struct {
+		name               string
+		conv               *internal.Conv
+		dbName             string
+		mockTableSetup     func(mock sqlmock.Sqlmock, tableName string, dbName string)
+		mockColumnSetup    func(mock sqlmock.Sqlmock, tableName, colName, dbName string)
+		checkSpecifics     func(t *testing.T, result map[string]utils.TableAssessmentInfo, tableID, colID string)
+		wantErrMsgContains string
+	}
+
+	testCases := []testCase{
```

Comment:
nit: These method is getting hard to understand. It would be good if this can be broken into different methods.